### PR TITLE
sso-proxy: command line tool to generate request signatures

### PR DESCRIPTION
**Pull request by benjsto**
## Problem
For testing signature validation code in other places, it's nice to be able to easily generate signatures. The prior way of doing this involved setting up a test service, making a request to it, and inspecting the request headers between SSO and the upstream to get the signature.

## Solution
Make a little command line tool within `sso-proxy` that takes URL, body, and headers to generate a request signature using the `request-signer`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ RUN go mod download
 COPY . .
 RUN cd cmd/sso-auth && go build -mod=readonly -o /bin/sso-auth
 RUN cd cmd/sso-proxy && go build -mod=readonly -o /bin/sso-proxy
+RUN cd cmd/sso-proxy/generate-request-signature && go build -mod=readonly -o /bin/sso-generate-request-signature
```
Comment (jphines) [inline Dockerfile:19]: Adding some documentation for the new binary would be helpful for others to figure out how to use the tool.

Comment (benjsto) [reply Dockerfile:19]: For sure, I'll add some documentation for it 👍

**Comment by jphines** [comment]
@benjsto wanted to follow up here - have you gotten some time to write those docs haven't pushed 'em yet?

**Comment by benjsto** [comment]
Sorry, didn't get a chance to do this yet -- will prioritize for this week!

File: Dockerfile (hunk 1 above)
Comment (benjsto) [reply Dockerfile:19]: Added a little markdown file in `/docs` explaining usage of the tool.

**Review by jphines — APPROVED** [review]

**Pull request merged**